backend/inspirehep/tools/utils.py

```diff
@@ -77,7 +77,10 @@ def find_references(references, requested_format):
         elif len(hits) > 1:
             errors.append({"ref": ref, "line": line, "type": "ambiguous"})
         else:
-            ret.append(hits[0]["_source"][display_format])
+            source_field = hits[0]["_source"]
+            ret.append(
+                source_field[display_format].replace(source_field["texkeys"][0], ref)
```

Comment:
this fails if there are no texkeys (might happen sometimes, because the legacy algorithm to assign texkeys might fail). I don't think you can do the substitution in that case, but you should handle it gracefully.

Reply:
I guess I should keep the Inspire texkey in those cases right? Even though it's not going to work when the users paste them into their documents. Should I show some kind of warning message in the UI?

Reply:
I think it's enough if you don't blow up, currently it affects only 30000 records, but maybe check with @StellaCh.

Reply:
It doesn't affect a big volume of records, but since we have a UI to show errors, displaying a message to users will be helpful to make them understand why this blows up and how they can avoid it.

Reply:
What message should I use so that they understand more or less the issue?